### PR TITLE
feat: set field as custom object label identifier in Object Detail

### LINKED ISSUE #3302: Settings/Objects/Object Detail - Fields section - Implement "Set as record text" menu item for Custom objects

### PR DESCRIPTION
**Pull request by thaisguigon**
Closes #3302

TODO:
- [x] prevent user from disabling the label identifier field on backend

### CODE REVIEW

**Comment by lucasbordeau** [comment]
There a strange behavior with setting the record label identifier on another field, should we address this in this ticket ? 


https://github.com/twentyhq/twenty/assets/26528466/ab1a86e1-dadd-4e58-8dc5-b22ffaecb813

**Comment by thaisguigon** [comment]
> There a strange behavior with setting the record label identifier on another field, should we address this in this ticket ?

Hi, thank you for the review!

I was planning to address the Tables side with this issue: https://github.com/twentyhq/twenty/issues/3303
However, I just opened a separate issue (https://github.com/twentyhq/twenty/issues/3502) for fixing what is displayed in the chip cell + link to the show page, as https://github.com/twentyhq/twenty/issues/3303 is more about the column's position and allowed actions!

File: packages/twenty-front/src/pages/settings/data-model/SettingsObjectFieldEdit.tsx [hunk 1]
```diff
@@ -203,15 +204,20 @@ export const SettingsObjectFieldEdit = () => {
...
-        <Section>
-          <H2Title title="Danger zone" description="Disable this field" />
-          <Button
-            Icon={IconArchive}
-            title="Disable"
-            size="small"
-            onClick={handleDisable}
-          />
-        </Section>
+        {!isLabelIdentifierField({
```
[the remark this answers is not in the document]
Comment (thaisguigon) [reply packages/twenty-front/src/pages/settings/data-model/SettingsObjectFieldEdit.tsx]: Done!

**Review by charlesBochet — APPROVED** [review]
LGTM, thank you!

**Pull request merged**